### PR TITLE
Enable resizable minimap panel

### LINKED ISSUE #1489: Make the minimap resizeable
**Is your feature request related to a problem? Please describe.**

It may be too big for some people or on some systems. Also mentioned in #1358 

**Describe the solution you'd like**

As title says

**Describe alternatives you've considered**

Big/normal/small UI modes

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1489

### CODE REVIEW

File: client/minimap_panel.h [hunk 1]
```diff
@@ -4,17 +4,32 @@
...
+class QItemSelection;
+class QListWidget;
+class QMouseEvent;
+class QObject;
+class QPaintEvent;
+class QPainter;
+class QPushButton;
+class QPixmap;
+class QResizeEvent;
+
```
Comment (lmoureaux) [inline client/minimap_panel.h:25]: Why do we need those?

**Review by lmoureaux — COMMENTED** [review]
Code only, I need to test if it remembers the size after closing and reopening (that code will always surprise me)

File: client/minimap_panel.h (hunk 1 above)
Comment (jwrober) [reply client/minimap_panel.h:25]: I don't think it remembers the size. I copied over from the messages widget, so might not need all of them.

Comment (jwrober) [reply client/minimap_panel.h]: I pushed another commit getting rid of that extra stuff that turns out I didn't need. I can confirm that the geometry is not saved between turns, same as messages widget. I am open to adding that to this PR for minimap_panel. Could use some direction on how to do that.

**Review by lmoureaux — APPROVED** [review]
I think we can got without remembering the size for now. What we have for the chat is broken anyway, and it would make sense to use a common solution.

**Pull request merged**